pkg/gofr/datasource/mongo/mongo_test.go

```diff
@@ -36,17 +36,155 @@ func Test_NewMongoClient(t *testing.T) {
 	assert.NotNil(t, client)
 }
 
+func TestGenerateMongoURI(t *testing.T) {
+	tests := []struct {
+		name          string
+		config        Config
+		expectedURI   string
+		expectedHost  string
+		expectedError string
+	}{
+		{
+			name: "Valid Config",
+			config: Config{
+				User:     "admin",
+				Password: "password",
```

Comment:
Could you please update the password to something that cont wins characters that need to be encoded

Something like


```suggestion
				Password: "p@##word:",
```


This way you will validate they are escaped